### PR TITLE
Take the result of image_get_resinfo as uniform if possible

### PR DESCRIPTION
**Pull request by Kai-AMD**

### CODE REVIEW

**Comment by JaxLinAMD** [comment]
retest this please

**Comment by amdvlk-admin** [comment]
### Test summary for commit 0911c43c5901b06273ce453a47e40394b5806711
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 712f76fee75d69b23a1ea8f6465752c3ccaaf9a2</li>
<li>PAL: dev b30561984a7b697ed7c709be76ff2e4d30532728</li>
<li>SPVGEN: dev b28e58f562591f335b9a3b5ae7e578a426df210b</li>
<li>XGL: dev 9bcc23fb29dab0b7ed1ebff17f3e99ae4da0bdf9</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev b3a0782762d195b57a2bebaabca24be90a35db2a</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/441035)</summary><ul><li>Built with branch Vulkan-CTS-1.2.0.2-fetch-sources-fix</li>
<details open><summary>Rhel 8.2, Gfx8</summary><ul>

<li>Passed: 146969/146969 (100.0%)</li>
<li>Failed: 0/146969 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 147013/147013 (100.0%)</li>
<li>Failed: 0/147013 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 147053/147053 (100.0%)</li>
<li>Failed: 0/147053 (0.0%)</li>
</ul></details>


</ul></details>

File: lgc/builder/ImageBuilder.cpp [hunk 1]
```diff
@@ -1331,6 +1337,15 @@ Value *ImageBuilder::CreateImageQuerySize(unsigned dim, unsigned flags, Value *i
...
 
+  if (!(flags & ImageFlagNonUniformImage)) {
+    for (unsigned i = 0; i < sizeComponentCount; i++) {
+      Value *index = getInt32(i);
+      Value *element = CreateExtractElement(intResInfo, index, instName);
+      element = CreateIntrinsic(Intrinsic::amdgcn_readfirstlane, {}, element);
+      intResInfo = CreateInsertElement(intResInfo, element, index);
+    }
+  }
+
```
Comment (nhaehnle) [inline lgc/builder/ImageBuilder.cpp:1348]: I don't think this is correct: `lod` could be a non-uniform value, and in that case the result would also be non-uniform.

**Comment by nhaehnle** [comment]
This has motivated me to open issues #844 and #845. Issue #844 is quite complex and big, but issue #845 could be a better performance improvement for what this pull request is addressing. Do you think you could look into that issue?

File: lgc/builder/ImageBuilder.cpp (hunk 1 above)
Comment (Kai-AMD) [reply lgc/builder/ImageBuilder.cpp:1348]: Yes, you are right. I will try to check it.

**Comment by Kai-AMD** [comment]
> This has motivated me to open issues #844 and #845. Issue #844 is quite complex and big, but issue #845 could be a better performance improvement for what this pull request is addressing. Do you think you could look into that issue?

Yes, I'd love to take it, let's go on the discussion in #845.

File: lgc/builder/ImageBuilder.cpp (hunk 1 above)
Comment (Kai-AMD) [reply lgc/builder/ImageBuilder.cpp:1348]: @nhaehnle, do you think if we still need to go ahead on this pull request? Or just move to implement #845.

Comment (nhaehnle) [reply lgc/builder/ImageBuilder.cpp:1348]: If #845 is done we don't need this anymore.

**Comment by Kai-AMD** [comment]
Since #845 will be implemented and #843 is no longer necessary, I close this pull request.

**Pull request closed without merging**